test/testFortranCreateRebuildMPs.f90

```diff
@@ -0,0 +1,318 @@
+    do i = 5,numMPs
+      mp2Elm(i) = i-2 !i=5 leads to mp2Elm(5)=3 (5-th MP in 3-rd element)
+                      !i=numMPs leads to mp2Elm(numMPs=nCells+2)=numMPs-2=nCells
+    end do
+  
+    allocate(mpsPerElm(nCells))
+    mpsPerElm = 1 !all elements have 1 MP and some changed below
+    mpsPerElm(1) = 0 !1st element has 0 MPs
+    mpsPerElm(2) = 2 !2nd element has 2 MPs
+    mpsPerElm(3) = 2 !3rd element has 2 MPs 
+
+    call polympo_createMPs(mpMesh,nCells,numMPs,c_loc(mpsPerElm),c_loc(mp2Elm),c_loc(isMPActive))
+
+    !set mp positions
+    do i = 1,numMPs
+      mpPosition(1,i) = i+ptOne
+      mpPosition(2,i) = numMPs+i+ptOne
+      mpPosition(3,i) = (2*numMPs)+i+ptOne
+    end do
+
+    call polympo_setMPPositions(mpMesh,nDims,numMPs,c_loc(mpPosition))
+    mpPosition = 42
+    call polympo_getMPPositions(mpMesh,nDims,numMPs,c_loc(mpPosition))
+    do i = 1,numMPs
+      if(isMPActive(i) .eq. MP_ACTIVE) then
+        call assert(epsilonDiff(mpPosition(1,i),i+ptOne), "x position of MP does not match")
+        call assert(epsilonDiff(mpPosition(2,i),numMPs+i+ptOne), "y position of MP does not match")
+        call assert(epsilonDiff(mpPosition(3,i),(2*numMPs)+i+ptOne), "z position of MP does not match")
+      endif
+    end do
+    
+    mp2Elm = -99 !override values and then use get function below
+    call polympo_getMPCurElmID(mpMesh,numMPs,c_loc(mp2Elm))
+    call assert(mp2Elm(1) .eq. 2, "wrong element ID for MP 1")
+    call assert(mp2Elm(2) .eq. 3, "wrong element ID for MP 2")
+    call assert(mp2Elm(3) .eq. 2, "wrong element ID for MP 3")
+    !mp2Elm(4) is not needed/used since 4-th MP is inactive
+    do i = 5,numMPs
+      call assert(mp2Elm(i) .eq. i-2, "wrong element ID for i'th MP")
+    end do
+    !test end
+
+    !deallocate MP variables
+    deallocate(mpsPerElm)
+end subroutine
+
+subroutine rebuildTests(mpMesh, numMPs, mp2Elm, isMPActive, mpPosition)
+    use :: polympo
+    use iso_c_binding
+    implicit none
+    type(c_ptr):: mpMesh
+    integer :: numMPs, i, j, numMPsLarger
+    integer, dimension(:), pointer :: mp2Elm, addedMPMask, isMPActive, mp2ElmFromPMPO
+    integer, dimension(:), pointer :: mp2ElmLarger, addedMPMaskLarger, mp2ElmFromPMPOLarger, isMPActiveLarger
+    real(kind=MPAS_RKIND), dimension(:,:), pointer :: mpPosition, mpPositionFromPMPO
+    integer, parameter :: nDims = 3
+    integer, parameter :: MP_ACTIVE = 1
+    integer, parameter :: MP_INACTIVE = 0
+    integer, parameter :: MP_DELETE_ELM_ID = -1
+
+    ! TEST: adding MP
+    ! Necessary pre-conditions for test
+    call assert(numMPs >= 4, "not enough MPs for test")
+    call assert(isMPActive(4) == MP_INACTIVE, "mp2Elm = 4 is active")
+    ! PREPARE DATA
+    allocate(addedMPMask(numMPs))
+    addedMPMask = MP_INACTIVE
+    isMPActive(4) = MP_ACTIVE
+    mp2Elm(4) = 7
+    addedMPMask(4) = MP_ACTIVE
+    mpPosition(1,4) = 1.2
+    mpPosition(2,4) = 2.2
+    mpPosition(3,4) = 3.2
```

Comment:
Use 1.2_MPAS_RKIND, 2.2_MPAS_RKIND and 3.2_MPAS_RKIND

Reply:
https://github.com/SCOREC/polyMPO/commit/bb04ba082488281d308d7df12e232b6744b0d99b